src/main/java/com/example/bookshop/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,22 @@
+package com.example.bookshop.security;
+
+import com.example.bookshop.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email)
+            throws UsernameNotFoundException {
+        return userRepository.findByEmail(email)
+                .orElseThrow(() -> new UsernameNotFoundException(
+                        "Can't find user with email: " + email));
```

Comment:
It would be better to place it in two lines
```suggestion
        return userRepository.findByEmail(email)
           .orElseThrow(() -> new UsernameNotFoundException("Can't find user with email: " + email));
```